File: DSA/ShatakshiMishra_590015008/Day6Question1.cpp
#include <iostream>
using namespace std;

int main() {
	int n;
	cin >> n;
	
	int nums[n+1];
	int calculated_sum = 0;
	for(int i=0;i<n+1;i++)
	{
	    cin >> nums[i];
	    calculated_sum+=nums[i];
	}
	
	int sum = (n*(n+1))/2;
	
	cout << calculated_sum - sum << endl;

}

File: DSA/Sauhard_Semalty_590013630/Day6Question1.py
import sys


def duplicate_number_from_range(lst: list[int], n: int) -> int:
    """Finds the duplicate element from the list containing
    numbers from 1 to n."""

    real_sum = n*(n + 1) // 2  # Arithmetic progression formula

    return sum(lst) - real_sum


def main() -> None:
    "The entry point of the program."

    try:
        n = int(input("Enter the size of the array: "))
    except ValueError:
        print("Invalid size!")
        sys.exit(-1)

    # Check if length is valid
    if n <= 0:
        print("Size must be greater 0.")
        sys.exit(-1)

    nums: list[int] = []

    print(f"+--Enter {n + 1} numbers between 1 to {n} with one duplicate--+")
    for _ in range(n + 1):
        try:
            num = int(input("> "))
        except ValueError:
            print("Invalid value!")
            sys.exit(-1)

        if not 0 <= num <= n:
            print(f"Number must be between 0 and {n} (inclusive).")
            sys.exit(-1)

        nums.append(num)

    print(duplicate_number_from_range(nums, n))


if __name__ == "__main__":
    main()
